Training/interface/DataLoaderReco_main.h

```diff
@@ -0,0 +1,264 @@
+          return tauTuple->get<std::vector<BranchType>>(prefix + variable);
+      }
+
+      void FillLabels(size_t tau_i,
+                      size_t n_classes)
+      {
+        auto& tau = const_cast<tau_tuple::Tau&>(tauTuple->data());
+        auto genLeptons = reco_tau::gen_truth::GenLepton::fromRootTuple(
+                            tau.genLepton_lastMotherIndex,
+                            tau.genParticle_pdgId,
+                            tau.genParticle_mother,
+                            tau.genParticle_charge,
+                            tau.genParticle_isFirstCopy,
+                            tau.genParticle_isLastCopy,
+                            tau.genParticle_pt,
+                            tau.genParticle_eta,
+                            tau.genParticle_phi,
+                            tau.genParticle_mass,
+                            tau.genParticle_vtx_x,
+                            tau.genParticle_vtx_y,
+                            tau.genParticle_vtx_z);
+        auto getVecRef = [&](size_t element) ->  Float_t&{
+            return data->y.at(tau_i*n_classes + element);
+            };
+        getVecRef(0) = genLeptons.nChargedHadrons();
+        getVecRef(1) = genLeptons.nNeutralHadrons();
+        getVecRef(2) = genLeptons.visibleP4().Pt();
+        getVecRef(3) = TMath::Power(genLeptons.visibleP4().M(),2);
+      }
+
+      template<typename PfType>
+      void FillPfSequence(Long64_t tau_i,
+                          const size_t n_sequence,
+                          const size_t n_features,
+                          const std::string& pref)
+      {
+        size_t start_index = n_sequence * n_features * tau_i;
+
+        std::vector<size_t> indices(tauTuple->get<std::vector<Float_t>>(pref+"pt").size());
+        std::iota(indices.begin(), indices.end(), 0);
+
+        // sort PfCands by pt
+        std::sort(indices.begin(), indices.end(), [&](size_t a, size_t b) {
+            return tauTuple->get<std::vector<Float_t>>(pref+"pt").at(a) >
+                   tauTuple->get<std::vector<Float_t>>(pref+"pt").at(b);
+        });
+        
+        Float_t empty_float;
+        size_t upper_index = std::min(n_sequence, indices.size());
+        for(size_t pfCand_i = 0; pfCand_i < upper_index; pfCand_i++)
+        {
+            auto getVecRef = [&](PfType _fe, Float_t value){
+                size_t _feature_idx = static_cast<size_t>(_fe);
+                if(_feature_idx < 0) return;
+                size_t index = start_index + n_features * pfCand_i + _feature_idx;
+                data->x.at(index) = 
+                    Scale<typename  FeaturesHelper<decltype(_fe)>::scaler_type>(
+                        _feature_idx, value, false
+                        );
+            };
+
+            size_t idx_srt = indices.at(pfCand_i);
+            getVecRef(PfType::pfCand_valid                ,1.0);
+            getVecRef(PfType::pfCand_pt                   ,getValue<Float_t>("pt",pref).at(idx_srt));
```

Comment:
NB. `getValue` by name is very slow... probably we should think about converting the dataset to `.hdf5` to improve the training speed. (no action required for this PR)

Reply:
so, should I change from the reference `getValue<Float_t>("eta",pref).at(idx_srt)` to `tau.eta[idx_srt]` here and everywhere + for pfCand?

Reply:
More like `tau.pfCand_eta[idx_srt]` etc. The idea of using prefix was to have the same code to read `lostTrack` and `pfCand` collections, but at the end also the scaling will be different, so I think it is more convenient to have explicit access to variables without using get & prefix.